packages/Chem/src/OCL-service/OCL-service.ts

```diff
@@ -0,0 +1,59 @@
+import * as DG from 'datagrok-api/dg';
+import {OCLServiceCall} from './consts';
+
+export class OCLService {
+  _workers: Worker[];
+  _threadCount: number = 4;
+  constructor() {
+    this._threadCount = Math.max((navigator.hardwareConcurrency || 6) - 2, 1);
+    this._workers = new Array(this._threadCount).fill(null)
+      .map(() => new Worker(new URL('OCL-worker', import.meta.url)));
+  }
+
+  private async _doParallel(
+    molCol:DG.Column, argList: any[], op: OCLServiceCall,
+  ): Promise<{[key: string]: Array<number>}> {
+    const molList = molCol.toList();
```

Comment:
Do we really need to create another array? Is it just for being able to use the `slice` method?

Reply:
yes